server/methods/accounts/accounts.js

```diff
@@ -141,65 +178,147 @@ function validateAddress(address) {
-      // `check` will not pass it.
-      if (!Reaction.hasAdminAccess()) {
-        throw new Meteor.Error(403, "Access denied");
-      }
+/**
+ * addressBookAdd
+ * @description add new addresses to an account
+ * @param {Object} address - address
+ * @param {String} [accountUserId] - `account.userId` used by admin to edit
+ * users
+ * @return {Object} with keys `numberAffected` and `insertedId` if doc was
+ * inserted
+ */
+export function addressBookAdd(address, accountUserId) {
+  check(address, Schemas.Address);
+  check(accountUserId, Match.Optional(String));
+  // security, check for admin access. We don't need to check every user call
+  // here because we are calling `Meteor.userId` from within this Method.
+  if (typeof accountUserId === "string") { // if this will not be a String -
+    // `check` will not pass it.
+    if (!Reaction.hasAdminAccess()) {
+      throw new Meteor.Error(403, "Access denied");
     }
-    this.unblock();
+  }
+  this.unblock();
 
-    const userId = accountUserId || Meteor.userId();
-    // required default id
-    if (!address._id) {
-      address._id = Random.id();
+  const userId = accountUserId || Meteor.userId();
+  // required default id
+  if (!address._id) {
+    address._id = Random.id();
+  }
+  // if address got shippment or billing default, we need to update cart
+  // addresses accordingly
+  if (address.isShippingDefault || address.isBillingDefault) {
+    const cart = Cart.findOne({ userId: userId });
+    // if cart exists
+    // First amend the cart,
+    if (typeof cart === "object") {
+      if (address.isShippingDefault) {
+        Meteor.call("cart/setShipmentAddress", cart._id, address);
+      }
+      if (address.isBillingDefault) {
+        Meteor.call("cart/setPaymentAddress", cart._id, address);
+      }
     }
-    // if address got shippment or billing default, we need to update cart
-    // addresses accordingly
-    if (address.isShippingDefault || address.isBillingDefault) {
-      const cart = Cart.findOne({ userId: userId });
-      // if cart exists
-      // First amend the cart,
-      if (typeof cart === "object") {
-        if (address.isShippingDefault) {
-          Meteor.call("cart/setShipmentAddress", cart._id, address);
+    // then change the address that has been affected
+    if (address.isShippingDefault) {
+      Accounts.update({
+        "userId": userId,
+        "profile.addressBook.isShippingDefault": true
+      }, {
+        $set: {
+          "profile.addressBook.$.isShippingDefault": false
         }
-        if (address.isBillingDefault) {
-          Meteor.call("cart/setPaymentAddress", cart._id, address);
+      });
+    }
+    if (address.isBillingDefault) {
+      Accounts.update({
+        "userId": userId,
+        "profile.addressBook.isBillingDefault": true
+      }, {
+        $set: {
+          "profile.addressBook.$.isBillingDefault": false
         }
-      }
-      // then change the address that has been affected
+      });
```

Comment:
While I support moving these into a more standard format, it is best practice to limit your PR to just changes that affect the issue you are trying to fix (it's in our docs and in the PR template). These changes make it difficult to see what is affecting this issue and what is "refactoring". It also makes it difficult for me to know what to test now.